src/file.toit

```diff
@@ -251,9 +259,80 @@ Deletes a file, given its name.
 delete name/string -> none:
   #primitive.file.unlink
 
-/*
+/**
 Renames a file or directory.
 Only works if the $to name is on the same filesystem.
 */
 rename from/string to/string -> none:
   #primitive.file.rename
+
+/**
+Creates a hard link from $source to a $target file.
+*/
+link --hard --source/string --target/string -> none:
+  if not hard: throw "INVALID_ARGUMENT"
+  link_ source target LINK_TYPE_HARD_
+
+/**
+Creates a soft link from $source to a $target file.
+*/
+link --file --source/string --target/string -> none:
+  if not file: throw "INVALID_ARGUMENT"
+  if is-directory target:
+    throw "Target is a directory"
+  link_ source target LINK_TYPE_SYMBOLIC_
+
+/**
+Creates a soft link from $source to a $target directory.
+*/
+link --directory --source/string --target/string -> none:
+  if not directory: throw "INVALID_ARGUMENT"
+  if is-file target:
+    throw "Target is a file"
+  if system.platform == system.PLATFORM-WINDOWS:
+    link_ source target LINK_TYPE_SYMBOLIC_WINDOWS_DIRECTORY_
+  else:
+    link_ source target LINK_TYPE_SYMBOLIC_
+
+/**
+Creates a symbolic link from $source to $target. This version of link requires
+  that the $target exists.
+It will automatically choose the correct type of link (file or directory) based
+  on the type of $target.
+*/
+link --source/string --target/string -> none:
+  if not stat target: throw "INVALID_ARGUMENT"
```

Comment:
"Invalid argument" is too hard to debug.
```suggestion
  if not stat target: throw "SOURCE NOT FOUND"
```

Reply:
How about "TARGET_NOT_FOUND"?